### PR TITLE
[Merged by Bors] - fix({bot_fix_style*,lint_and_suggest}.yml): don't use continue-on-error

### PR DESCRIPTION
**Pull request by bryangingechen**
These workflows had linting steps which were expected to fail with nonzero exit codes and then have those failures ignored by the (step-level) [`continue-on-error` setting](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error). This setting only allows the workflow to continue executing; the error still shows up as a red x in the step, and also causes the entire job to fail, unless [`continue-on-error` is set at the job level](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error), which we ended up doing for the "bot fix style" workflows.

With this setting active, the job will succeed regardless of any errors that occur, which has the downside that errors in other steps might not get noticed.

In #16301 the exit code was changed for `lint-style --fix` making the step-level `continue-on-error` setting unnecessary for those linting steps. In this PR I:
- add `|| true` to the end of other such linting steps so that their exit codes are also ignored, which let me
- remove all step-level `continue-on-error` settings, which let me
- remove all job-level `continue-on-error` settings

In summary, error reporting for these workflows will now be much more informative.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by grunweg — COMMENTED** [review]
Thanks for cleaning this up. And I'm glad the exit code change to lint-style --fix is useful!
maintainer merge

File: .github/workflows/bot_fix_style_comment.yaml [hunk 1]
```diff
@@ -70,14 +68,12 @@ jobs:
 
       - name: lint references.bib
         if: steps.user_permission.outputs.require-result == 'true'
-        continue-on-error: true  # allows the following `reviewdog` step to add GitHub suggestions
         run: |
-          ./scripts/lint-bib.sh
+          ./scripts/lint-bib.sh || true
```
Comment (Vierkantor) [inline .github/workflows/bot_fix_style_comment.yaml:72]: Let's retain the comment:
```suggestion
          # ignoring the return code allows the following `reviewdog` step to add GitHub suggestions
          ./scripts/lint-bib.sh || true
```

File: .github/workflows/bot_fix_style_comment.yaml [hunk 2]
```diff
@@ -70,14 +68,12 @@ jobs:
...
 
       - name: update {Mathlib, Tactic, Counterexamples, Archive}.lean
         if: steps.user_permission.outputs.require-result == 'true'
-        continue-on-error: true  # allows the following `reviewdog` step to add GitHub suggestions
-        run: lake exe mk_all
+        run: lake exe mk_all || true
```
Comment (Vierkantor) [inline .github/workflows/bot_fix_style_comment.yaml:76]:
```suggestion
        run: |
          # ignoring the return code allows the following `reviewdog` step to add GitHub suggestions
          lake exe mk_all || true
```

File: .github/workflows/bot_fix_style_review.yaml [hunk 3]
```diff
@@ -76,14 +74,12 @@ jobs:
 
       - name: lint references.bib
         if: steps.user_permission.outputs.require-result == 'true'
-        continue-on-error: true  # allows the following `reviewdog` step to add GitHub suggestions
         run: |
-          ./scripts/lint-bib.sh
+          ./scripts/lint-bib.sh || true
```
Comment (Vierkantor) [inline .github/workflows/bot_fix_style_review.yaml:78]:
```suggestion
          # ignoring the return code allows the following `reviewdog` step to add GitHub suggestions
          ./scripts/lint-bib.sh || true
```

File: .github/workflows/bot_fix_style_review.yaml [hunk 4]
```diff
@@ -76,14 +74,12 @@ jobs:
...
 
       - name: update {Mathlib, Tactic, Counterexamples, Archive}.lean
         if: steps.user_permission.outputs.require-result == 'true'
-        continue-on-error: true  # allows the following `reviewdog` step to add GitHub suggestions
-        run: lake exe mk_all
+        run: lake exe mk_all || true
```
Comment (Vierkantor) [inline .github/workflows/bot_fix_style_review.yaml:82]:
```suggestion
        run: |
          # ignoring the return code allows the following `reviewdog` step to add GitHub suggestions
          lake exe mk_all || true
```

File: .github/workflows/bot_fix_style_review_comment.yaml [hunk 5]
```diff
@@ -74,14 +72,12 @@ jobs:
 
       - name: lint references.bib
         if: steps.user_permission.outputs.require-result == 'true'
-        continue-on-error: true  # allows the following `reviewdog` step to add GitHub suggestions
         run: |
-          ./scripts/lint-bib.sh
+          ./scripts/lint-bib.sh || true
```
Comment (Vierkantor) [inline .github/workflows/bot_fix_style_review_comment.yaml:76]:
```suggestion
          # ignoring the return code allows the following `reviewdog` step to add GitHub suggestions
          ./scripts/lint-bib.sh || true
```

File: .github/workflows/bot_fix_style_review_comment.yaml [hunk 6]
```diff
@@ -74,14 +72,12 @@ jobs:
...
 
       - name: update {Mathlib, Tactic, Counterexamples, Archive}.lean
         if: steps.user_permission.outputs.require-result == 'true'
-        continue-on-error: true  # allows the following `reviewdog` step to add GitHub suggestions
-        run: lake exe mk_all
+        run: lake exe mk_all || true
```
Comment (Vierkantor) [inline .github/workflows/bot_fix_style_review_comment.yaml:80]:
```suggestion
        run: |
          # ignoring the return code allows the following `reviewdog` step to add GitHub suggestions
          lake exe mk_all || true
```

File: .github/workflows/lint_and_suggest_pr.yml [hunk 7]
```diff
@@ -46,9 +45,8 @@ jobs:
 
       # if you update this step (or its dependencies), please also update them in bot_fix_style_comment.yaml
       - name: lint references.bib
-        continue-on-error: true  # allows the following `reviewdog` step to add GitHub suggestions
         run: |
-          ./scripts/lint-bib.sh
+          ./scripts/lint-bib.sh || true
```
Comment (Vierkantor) [inline .github/workflows/lint_and_suggest_pr.yml:49]:
```suggestion
          # ignoring the return code allows the following `reviewdog` step to add GitHub suggestions
          ./scripts/lint-bib.sh || true
```

File: .github/workflows/lint_and_suggest_pr.yml [hunk 8]
```diff
@@ -75,8 +73,7 @@ jobs:
 
       # if you update this step (or its dependencies), please also update them in bot_fix_style_comment.yaml
       - name: update {Mathlib, Tactic, Counterexamples, Archive}.lean
-        continue-on-error: true  # allows the following `reviewdog` step to add GitHub suggestions
-        run: lake exe mk_all
+        run: lake exe mk_all || true
```
Comment (Vierkantor) [inline .github/workflows/lint_and_suggest_pr.yml:76]:
```suggestion
        run: 
          # ignoring the return code allows the following `reviewdog` step to add GitHub suggestions
          lake exe mk_all || true
```

**Review by Vierkantor — APPROVED** [review]
Thanks!

bors d+

**Comment by bryangingechen** [comment]
Thanks for the suggestions!
bors r+
